src/plugins/intel_npu/src/al/include/npu_private_properties.hpp

```diff
@@ -151,6 +151,43 @@ inline std::ostream& operator<<(std::ostream& out, const ElfCompilerBackend& fmt
     return out;
 }
 
+/**
+ * @brief [Only for NPU Plugin]
+ * Type: String. Default is "AUTO".
+ * This option is added for enabling batching on plugin.
+ * Possible values: "AUTO", "COMPILER", "PLUGIN".
+ */
+enum class BatchMode {
+    AUTO = 0,
+    COMPILER = 1,
+    PLUGIN = 2,
+};
+
+/**
+ * @brief Prints a string representation of ov::intel_npu::BatchMode to a stream
+ * @param out An output stream to send to
+ * @param fmt A value for batching on plugin to print to a stream
+ * @return A reference to the `out` stream
+ * @note Configuration API v 2.0
+ */
+inline std::ostream& operator<<(std::ostream& out, const BatchMode& fmt) {
+    switch (fmt) {
+    case BatchMode::AUTO: {
+        out << "AUTO";
+    } break;
+    case BatchMode::COMPILER: {
+        out << "COMPILER";
+    } break;
+    case BatchMode::PLUGIN: {
+        out << "PLUGIN";
+    } break;
+    default:
+        out << static_cast<uint32_t>(fmt);
```

Comment:
We throw an error in case the choice is wrong much earlier in the code.

Reply:
I see. Nonetheless, I don't see much value registering the raw value here.

Also, as a general rule of thumb I tend to follow - it might be better to design the code in a more modularized manner, without too many implicit assumptions on how the outside code works at a given time. The main reason for this is: there are multiple contributors to the project which may break the flow without being aware of the referenced assumptions. That's perhaps the main reasons why assertions exist - to validate that your assumptions are still holding.

This is a nitpick anyway and I don't wish to waste more valuable development time on this - feel free to mark this as resolved once you are satisfied with the approach.